float32[] trajectoryX
float32[] trajectoryY